js/attendeeContact.js

```diff
@@ -98,29 +105,29 @@ function getParameterByName(name, callback)
 // Increment the badge number and set the date/time stamp fields
 // Return true or false (Depending upon success.
 function incrementBadge() {
-    var numActiveBadgesFieldId = 7;
-    var pickupDateFieldIds = [ 9, 11, 13, 15 ];
-    var pickupTimeFieldIds = [ 10, 12, 14, 16 ];
 
     console.log("incrementBadge Received a request and is processing it now");
-    var checkFieldValue = document.getElementsByName('attendee.customDataList[' + numActiveBadgesFieldId + '].value')[0].value;
-    var numActive = parseInt(document.getElementsByName('attendee.customDataList[' + numActiveBadgesFieldId + '].value')[0].value);
-    var elNumberActiveBadges = document.getElementsByName('attendee.customDataList[' + numActiveBadgesFieldId + '].value')[0];
+
+    var elNumberActiveBadges = $("label:contains('Number of Active Badges'):first").next().find("input");
+    var numActive = elNumberActiveBadges.val() === '' ? 0 : parseInt(elNumberActiveBadges.val());
+
     var saveButton = document.getElementsByName('save')[0];
 
     var pickupDates = new Array();
     var elpickupDates = new Array();
-    pickupDateFieldIds.forEach(function(index){
-        pickupDates.push(document.getElementsByName('attendee.customDataList[' + index + '].value')[0].value);
-        elpickupDates.push(document.getElementsByName('attendee.customDataList[' + index + '].value')[0]);
-    });
+    for( var i = 1; i < 5; i++ )
+    {
+      pickupDates.push($("label:contains('Pickup Date " + i + "'):first").next().find("input").val());
+      elpickupDates.push($("label:contains('Pickup Date " + i + "'):first").next().find("input"));
+    }
 
     var pickupTimes = new Array();
     var elpickupTimes = new Array();
-    pickupTimeFieldIds.forEach(function(index){
-        pickupTimes.push(document.getElementsByName('attendee.customDataList[' + index + '].value')[0].value);
-        elpickupTimes.push(document.getElementsByName('attendee.customDataList[' + index + '].value')[0]);
-    });
+    for( var i = 1; i < 5; i++ )
+    {
+        pickupTimes.push($("label:contains('Pickup Time " + i + "'):first").next().find("input").val());
+        elpickupTimes.push($("label:contains('Pickup Time " + i + "'):first").next().find("input"));
+    }
 

```

Comment:
I wonder here...
the old code was a forEach which would have been as many items in 'pickupTimeFieldIds' as there are.  Your code hardcodes 1 to 5. Is that because it is hard coded somewhere else you are also matching?  This is not a serious issue but just want to make sure we are not limiting ourselves where it was not limited before.

Reply:
It's just that the Attendee forms only contain 4 fields each for Pickup Time/Date.  That limit is based on Reg policy of only allowing a badge to be re-printed 3 times (for 4 badges total) before we make them buy a new badge instead of a reprint.